### PR TITLE
Instructions for upgrading depending on the current installation

### PR DESCRIPTION
**Pull request by inercia**
## Description

Depending on the current installation found in the cluster, we should give some customized upgrade instructions.

### CODE REVIEW

File: docs/topics/install/upgrading.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
 # Upgrading Ambassador Edge Stack
 
-Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process is necessary to upgrade Ambassador Edge Stack. If you're using the YAML files supplied by Datawire, you'll be able to upgrade simply by repeating the following `kubectl apply` commands.
+Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process
+is necessary to upgrade Ambassador Edge Stack.
+
+First, determine the installation method by running some `kubectl` command:
+
+* If the following command returns some resources then you should
```
Comment (bnfb) [inline docs/topics/install/upgrading.md:8]: We need to have examples of the output that one would see in yes and no cases. Not just "returns some resources": please give examples of what the output would look like in the yes and no cases.

File: docs/topics/install/upgrading.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+$ kubectl get deployment -n ambassador -l 'app.kubernetes.io/name=ambassador,app.kubernetes.io/managed-by in (amb-oper,amb-oper-manifest,amb-oper-helm,amb-oper-azure)' 
+```
+
+* If the following command returns some resources you should
+[with the help of Helm](https://www.getambassador.io/docs/1.3/topics/install/helm/#migrating-to-the-ambassador-edge-stack):
+```commandline
+$ kubectl get deployment -n ambassador -l 'app.kubernetes.io/name=ambassador'
+```
+
+* Finally, if the following command returns some resources you should upgrade with manifests as
```
Comment (khussey) [inline docs/topics/install/upgrading.md:20]: Finally, if you installed using manifests, simply run the commands in the following section. To verify whether manifests were used to install Ambassador Edge Stack, run the following command to see if it returns resources:

File: docs/topics/install/upgrading.md [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+described in the following section:
+
+```commandline
+$ kubectl get deployment -n ambassador -l 'product=aes'
+```
+
+If none of these command show resources, you probably have an old installation and you should follow
```
Comment (khussey) [inline docs/topics/install/upgrading.md:27]: If none of the commands above return resources, ...

File: docs/topics/install/upgrading.md [hunk 4]
```diff
@@ -1,6 +1,36 @@
...
+the instructions for [upgrading to Ambassador Edge Stack](https://www.getambassador.io/docs/1.3/topics/install/upgrade-to-edge-stack/).
+
+## Upgrading an installation with manifests
+
+If you're using the YAML files supplied by Datawire, you'll be able to upgrade simply by repeating
+the following `kubectl apply` commands.
 
 First, determine if Kubernetes has RBAC enabled:
```
Comment (nbkrause) [inline docs/topics/install/upgrading.md:36]: Let's take this opportunity to remove this check for RBAC.

No modern cluster has rbac disabled unless effort was put in on the part of the provisioner to do so. This step causes more confusion than help.

File: docs/topics/install/upgrading.md [hunk 5]
```diff
@@ -1,6 +1,36 @@
 # Upgrading Ambassador Edge Stack
 
-Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process is necessary to upgrade Ambassador Edge Stack. If you're using the YAML files supplied by Datawire, you'll be able to upgrade simply by repeating the following `kubectl apply` commands.
+Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process
+is necessary to upgrade Ambassador Edge Stack.
+
+First, determine the installation method by running some `kubectl` command:
```
[the remark this answers is not in the document]
Comment (inercia) [reply docs/topics/install/upgrading.md:6]: Changed.

File: docs/topics/install/upgrading.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (inercia) [reply docs/topics/install/upgrading.md:8]: Changed.

Comment (inercia) [reply docs/topics/install/upgrading.md:8]: Done.

File: docs/topics/install/upgrading.md [hunk 6]
```diff
@@ -1,6 +1,36 @@
...
+
+* If the following command returns some resources then you should
+[upgrade the Operator](https://www.getambassador.io/docs/latest/topics/install/aes-operator/#updates-by-the-operator):
+```commandline
+$ kubectl get deployment -n ambassador -l 'app.kubernetes.io/name=ambassador,app.kubernetes.io/managed-by in (amb-oper,amb-oper-manifest,amb-oper-helm,amb-oper-azure)' 
+```
+
+* If the following command returns some resources you should
```
[the remark this answers is not in the document]
Comment (inercia) [reply docs/topics/install/upgrading.md:16]: @khussey This method also applies to `edgectl install`, so I have changed it slightly.

File: docs/topics/install/upgrading.md (hunk 2 above)
Comment (inercia) [reply docs/topics/install/upgrading.md:24]: Done

File: docs/topics/install/upgrading.md [hunk 7]
```diff
@@ -1,27 +1,49 @@
 # Upgrading Ambassador Edge Stack
 
-Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process is necessary to upgrade Ambassador Edge Stack. If you're using the YAML files supplied by Datawire, you'll be able to upgrade simply by repeating the following `kubectl apply` commands.
-
-First, determine if Kubernetes has RBAC enabled:
+Since Ambassador Edge Stack's configuration is entirely stored in Kubernetes resources, no special process
+is necessary to upgrade Ambassador Edge Stack.
+
+The steps to upgrade depend on the method that used used to install Ambassador Edge Stack, as indicated below.
```
Comment (khussey) [inline docs/topics/install/upgrading.md:6]: "used used" -> "was used"

File: docs/topics/install/upgrading.md [hunk 8]
```diff
@@ -1,27 +1,49 @@
...
+
+* If you installed using the Operator, then you'll need to [use the Operator to perform the upgrade](https://www.getambassador.io/docs/latest/topics/install/aes-operator/#updates-by-the-operator).
```
Comment (khussey) [inline docs/topics/install/upgrading.md:8]: Please make sure referenced to other documentation pages are relative, i.e., 'aes-operator/#updates-by-the-operator' in this case.

File: docs/topics/install/upgrading.md [hunk 9]
```diff
@@ -1,27 +1,49 @@
...
+```commandline
+$ kubectl get deployment -n ambassador -l 'app.kubernetes.io/name=ambassador,app.kubernetes.io/managed-by in (amb-oper,amb-oper-manifest,amb-oper-helm,amb-oper-azure)' 
+NAME               READY   UP-TO-DATE   AVAILABLE   AGE
+ambassador         1/1     1            1           ...
+```
 
-```shell
-kubectl cluster-info dump --namespace kube-system | grep authorization-mode
+* If you installed using the Helm chart or `edgectl install`, then you should
+[upgrade with the help of Helm](https://www.getambassador.io/docs/1.3/topics/install/helm/#migrating-to-the-ambassador-edge-stack).
```
Comment (khussey) [inline docs/topics/install/upgrading.md:18]: Needs to be a relative reference.

File: docs/topics/install/upgrading.md [hunk 10]
```diff
@@ -1,27 +1,49 @@
...
+manifests were used to install Ambassador Edge Stack, run the following command to see if it returns resources:
+```commandline
+$ kubectl get deployment -n ambassador -l 'product=aes'
+NAME               READY   UP-TO-DATE   AVAILABLE   AGE
+ambassador         1/1     1            1           ...
+```
 
-If RBAC is enabled:
+If none of the commands above return resources, you probably have an old installation and you should follow
+the instructions for [upgrading to Ambassador Edge Stack](https://www.getambassador.io/docs/1.3/topics/install/upgrade-to-edge-stack/).
```
Comment (khussey) [inline docs/topics/install/upgrading.md:35]: Reference needs to be relative.

**Pull request merged**